src/crimson/net/Connection.h

```diff
@@ -15,17 +15,18 @@
 #pragma once
 
 #include <queue>
-#include <boost/smart_ptr/intrusive_ref_counter.hpp>
 #include <seastar/core/future.hh>
+#include <seastar/core/shared_ptr.hh>
 
 #include "Fwd.h"
 
 namespace ceph::net {
 
 using seq_num_t = uint64_t;
+using SharedPtr = seastar::shared_ptr<seastar::shared_ptr_count_base>;
 
-class Connection : public boost::intrusive_ref_counter<Connection,
-						       boost::thread_unsafe_counter> {
+class Connection : public seastar::enable_shared_from_this<Connection> {
+  SharedPtr priv;
```

Comment:
i don't think this change is related to the enabling of x-core communication. could you extract it into its own commit?

Reply:
splitted to 3cb78640d0fb831f903e6d3f65ba13c0600b9052